### PR TITLE
Update protocol schema

### PR DESCRIPTION
**Pull request by Jasopaum**

### CODE REVIEW

File: protobuf/syft_proto/execution/v1/protocol.proto [hunk 1]
```diff
@@ -4,16 +4,16 @@ package syft_proto.execution.v1;
...
-
-message PlanAssignment {
-    syft_proto.types.syft.v1.Id plan_id = 1;
-    syft_proto.types.syft.v1.Id worker_id = 2;
-}
+    string name = 2;
+    map<string, syft_proto.execution.v1.Role> roles = 3;
+    repeated string input_repartition = 4;
+    repeated string output_repartition = 5;
+    bool include_state = 6;
```
Comment (karlhigley) [inline protobuf/syft_proto/execution/v1/protocol.proto]: Is it possible that some `Roles` will have `State` and others won't? It didn't make any difference for `Plans`, but wondering if we should move `State` into `Roles`.

File: protobuf/syft_proto/execution/v1/protocol.proto [hunk 2]
```diff
@@ -4,16 +4,16 @@ package syft_proto.execution.v1;
...
+    bool is_built = 7;
```
Comment (karlhigley) [inline protobuf/syft_proto/execution/v1/protocol.proto]: Can an unbuilt `Protocol` be serialized? I guess that train of thought applies to `Plans` too. 🤔

File: protobuf/syft_proto/execution/v1/protocol.proto [hunk 3]
```diff
@@ -4,16 +4,16 @@ package syft_proto.execution.v1;
...
-}
-
-message PlanAssignment {
-    syft_proto.types.syft.v1.Id plan_id = 1;
-    syft_proto.types.syft.v1.Id worker_id = 2;
-}
+    string name = 2;
+    map<string, syft_proto.execution.v1.Role> roles = 3;
+    repeated string input_repartition = 4;
+    repeated string output_repartition = 5;
```
Comment (karlhigley) [inline protobuf/syft_proto/execution/v1/protocol.proto]: This sounds like a thing that is probably useful, but I don't quite follow what it's used for yet. Fill me in?

File: protobuf/syft_proto/execution/v1/protocol.proto (hunk 1 above)
Comment (Jasopaum) [reply protobuf/syft_proto/execution/v1/protocol.proto]: Yeah, I think we need some thinking around State for Protocols (how to specify them, etc.). But I think it should be doable to move them into Roles. I don't think a Protocol could have a global State, could it?

File: protobuf/syft_proto/execution/v1/protocol.proto (hunk 2 above)
Comment (Jasopaum) [reply protobuf/syft_proto/execution/v1/protocol.proto]: Not at all, I thought the same thing while writing that! I can remove it for both Plan and Protocol.
(I think you made me shy about doing lots of refactoring/cleaning in a single PR 😄)

File: protobuf/syft_proto/execution/v1/protocol.proto (hunk 3 above)
Comment (Jasopaum) [reply protobuf/syft_proto/execution/v1/protocol.proto]: It's here to specify which Protocol args belong to which Role.
For instance, if we write protocol(a, b, c), we need to know to which role a, b, and c should go

File: protobuf/syft_proto/execution/v1/protocol.proto (hunk 1 above)
Comment (karlhigley) [reply protobuf/syft_proto/execution/v1/protocol.proto]: Not sure. Maybe the initial model for secure model aggregation?

Comment (Jasopaum) [reply protobuf/syft_proto/execution/v1/protocol.proto]: Maybe. But let's do that when we have an idea of how to solve the problem?

Comment (karlhigley) [reply protobuf/syft_proto/execution/v1/protocol.proto]: So resolved that we will assume there is no global state until proven otherwise. 👍

**Review by karlhigley — APPROVED** [review]

**Pull request merged**